tests/test-verify-runner-selection.sh

```
@@ -0,0 +1,200 @@
+
+  printf '{"scripts":{"test":"jest --ci"},"devDependencies":{"jest":"^29"}}' > "$_fx/package.json"
+  [ "$(_pick "$_fx")" = "jest" ] \
+    && ok "a declared jest script still selects jest (no over-correction)" \
+    || bad "a project that really does use jest no longer selects it"
+  /bin/rm -rf "$_fx" 2>/dev/null
+else
+  bad "could not create a fixture dir (assertions 8a-8c did not run)"
+fi
+
+# --- 9. Syntax --------------------------------------------------------------
+bash -n "$SRC" 2>/dev/null && ok "verify.sh parses" || bad "verify.sh has a syntax error"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)